### PR TITLE
Fix sigalarm for windows

### PR DESCRIPTION
**Pull request by irux**

### CODE REVIEW

File: kpops/component_handlers/kafka_connect/timeout.py [hunk 1]
```diff
@@ -14,17 +15,23 @@ def timeout(func: Callable[..., T], *, secs: int = 0) -> T | None:
...
+        task = asyncio.create_task(coro)
+        if secs == 0:
+            return await task
+        else:
+            return await asyncio.wait_for(task, timeout=secs)
 
     # set the timeout handler
-    signal.signal(signal.SIGALRM, handler)
-    signal.alarm(secs)
+    # signal.signal(signal.SIGALRM, handler)
```
Comment (raminqaf) [inline kpops/component_handlers/kafka_connect/timeout.py:27]: Remove old commented code please

File: kpops/component_handlers/kafka_connect/timeout.py [hunk 2]
```diff
@@ -14,17 +15,23 @@ def timeout(func: Callable[..., T], *, secs: int = 0) -> T | None:
     :param secs: The timeout in seconds
     """
 
-    def handler(signum, frame) -> NoReturn:
-        raise TimeoutError()
+    async def main_supervisor(func, secs):
+        coro = asyncio.to_thread(func)
```
Comment (raminqaf) [inline kpops/component_handlers/kafka_connect/timeout.py:19]: Can you give this variable a more meaningful name please?

**Review by raminqaf — CHANGES_REQUESTED** [review]

**Review by raminqaf — CHANGES_REQUESTED** [review]

**Comment by irux** [comment]
Yes! for example now all the connector test that depends on the timeout function works https://github.com/bakdata/kpops/blob/fix/windows/tests/component_handlers/kafka_connect/test_connect_wrapper.py#L122

**Review by sujuka99 — APPROVED** [review]
LGTM!

Please wait for another approval, too.

**Review by philipp94831 — APPROVED** [review]

**Review by disrupted — CHANGES_REQUESTED** [review]

File: kpops/component_handlers/kafka_connect/timeout.py [hunk 3]
```diff
@@ -14,17 +15,19 @@ def timeout(func: Callable[..., T], *, secs: int = 0) -> T | None:
     :param secs: The timeout in seconds
     """
 
-    def handler(signum, frame) -> NoReturn:
-        raise TimeoutError()
+    async def main_supervisor(func, secs):
```
[the remark this answers is not in the document]
Comment (irux) [reply kpops/component_handlers/kafka_connect/timeout.py]: @disrupted done!

**Pull request merged**